src/OpenSage.Game/Data/Wav/WavFile.cs

```diff
@@ -65,51 +135,118 @@ public sealed class WavFile
 
-        public void Parse(BinaryReader reader)
+        private int ByteToInt16(byte[] packed)
         {
-            _header = RiffHeader.Parse(reader);
+            // This is always little endian, unlike the C# builtin for unpacking a byte array.
+            return (packed[1] << 8) | packed[0];
+        }
 
-            if (new string(_header.ChunkId) != "RIFF" ||
-                new string(_header.Format) != "WAVE")
+        private byte[] ParseMicrosoft(BinaryReader reader, int size)
+        {
+            return reader.ReadBytes(size);
+        }
+        
+        private byte[] ParseDviAdpcm(BinaryReader reader, int size, int samplesPerBlock)
+        {
+            var imaIndexTable = new int[16] {
+              -1, -1, -1, -1, 2, 4, 6, 8,
+              -1, -1, -1, -1, 2, 4, 6, 8
+            };
+            var imaStepTable = new int[89] {
+              7, 8, 9, 10, 11, 12, 13, 14, 16, 17,
+              19, 21, 23, 25, 28, 31, 34, 37, 41, 45,
+              50, 55, 60, 66, 73, 80, 88, 97, 107, 118,
+              130, 143, 157, 173, 190, 209, 230, 253, 279, 307,
+              337, 371, 408, 449, 494, 544, 598, 658, 724, 796,
+              876, 963, 1060, 1166, 1282, 1411, 1552, 1707, 1878, 2066,
+              2272, 2499, 2749, 3024, 3327, 3660, 4026, 4428, 4871, 5358,
+              5894, 6484, 7132, 7845, 8630, 9493, 10442, 11487, 12635, 13899,
+              15289, 16818, 18500, 20350, 22385, 24623, 27086, 29794, 32767
+            };
+            int blockSize = (4 + (samplesPerBlock - 1) / 2);
+            if (size % blockSize != 0)
             {
-                throw new InvalidDataException("Invalid .wav file!");
+                throw new InvalidDataException("Invalid .wav DVI ADPCM data!");
             }
-
-            _format = WaveFormat.Parse(reader);
-
-            if (new string(_format.SubChunkID)!="fmt ")
+            int numBlocks = size / blockSize;
+            var buffer = new byte[samplesPerBlock * 2 * numBlocks];
+            using (var memoryStream = new MemoryStream(buffer))
+            using (var writer = new BinaryWriter(memoryStream))
             {
-                throw new InvalidDataException("Invalid .wav file!");
+                for (int i = 0; i < numBlocks; i++)
+                {
+                    int sample = reader.ReadInt16();
+                    int stepTableIndex = reader.ReadByte();
+                    byte reserved = reader.ReadByte(); // unused, commonly 0
+                    int step = imaStepTable[stepTableIndex];
+                    writer.Write((Int16) sample);
+                    for (int j = 0; j < (samplesPerBlock - 1) / 2; j++)
+                    {
+                        byte packed = reader.ReadByte();
+                        byte nibble_low = (byte) (packed & 0xF);
+                        byte nibble_high = (byte) (packed >> 4);
+                        foreach (var nibble in new byte[] { nibble_low , nibble_high })
```

Comment:
Also this part is quite inefficient since it always creates a List on the Heap. As @paavohuhtala pointed out to me C# added a stackalloc feature that allocates memory on the stack, which works like this:
```csharp
Span<byte> nibbles = stackalloc byte[2];
nibbles[0] = nibble_low;
nibbles[1] = nibble_high;
```

This should be much faster and much lower overhead. See [here](http://adamsitnik.com/Span/)  for more information about Span<T> with stackalloc feature. I tested this code locally and it does work